backend/app/models/case.rb

```diff
@@ -93,6 +95,12 @@ def attachment_count
   end
 
   private
+
+    def set_unique_code
+      highest_code = Case.maximum(:code).to_s.sub("CODE-", '').to_i
```

Comment:
Consider handling the case where the `Case.maximum(:code)` returns `nil`, which might happen if there are no cases in the database yet. [important]